src/Eagle.ts

```diff
@@ -730,6 +730,18 @@ export class Eagle {
         this.rightWindow().mode(Eagle.RightWindowMode.Hierarchy)
     }
 
+    changeRightWindowMode(requestedMode:Eagle.RightWindowMode) : void {
+        this.rightWindow().mode(requestedMode)
+        this.rightWindow().shown(true); 
+
+        //if we are intentionally switching to the hierary, then we do not want the right window mode to switch to the inspector when we change node selection.
+        if(requestedMode===Eagle.RightWindowMode.Hierarchy){
```

Comment:
This could be one line:

this.hierarchyMode(requestedMode === Eagle.RightWindowMode.Hierarchy);